_studio/mfx_lib/encode_hw/h264/src/mfx_h264_encode_hw.cpp

```diff
@@ -733,6 +733,65 @@ void ImplementationAvc::DestroyDanglingCmResources()
         }
     }
 }
+/*Class for modifying and restoring VideoParams.
+  Needed for initialization some components */
+class ModifiedVideoParams
+{
+public:
+    ModifiedVideoParams() :
+        m_bInit(false),
+        m_RateControlMethod(0),
+        m_LookAheadDepth(0),
+        m_MaxKbps(0)
+    {}
+    void ModifyForDDI(MfxVideoParam & par, bool bSWBRC)
+    {
+        if (!bSWBRC)
+            return;
+        if (!m_bInit)
+            SaveParams(par);
+        // in the case of SWBRC driver works in CQP mode
+        par.mfx.RateControlMethod = MFX_RATECONTROL_CQP;
+    }
+    void ModifyForBRC(MfxVideoParam & par, bool bSWBRC)
+    {
+        mfxExtCodingOption2 & extOpt2 = GetExtBufferRef(par);
+        if (!(bSWBRC && extOpt2.MaxSliceSize))
+            return;
+        if (!m_bInit)
+            SaveParams(par);
+        // int the case of MaxSliceSize BRC works in VBR mode (instead LA)
+        par.mfx.RateControlMethod = MFX_RATECONTROL_VBR;
+        par.mfx.MaxKbps = par.mfx.TargetKbps * 2;
```

Comment:
VBR is define with single bitrate. Now it is based on mfx.MaxKbps. Better use par.mfx.MaxKbps = par.mfx.TargetKbps